pkg/R/SQLContext.R

```diff
@@ -218,20 +218,74 @@ uncacheTable <- function(sqlCtx, tableName) {
   callJMethod(sqlCtx, "uncacheTable", tableName)
 }
 
-
 #' Clear Cache
 #'
 #' Removes all cached tables from the in-memory cache.
 #'
 #' @param sqlCtx SQLContext to use
+#' clearCache(sqlCtx)
+#' }
+
+clearCache <- function(sqlCtx) {
+  callJMethod(sqlCtx, "clearCache")
+}
+
+#' Load an DataFrame
+#'
+#' Returns the dataset in a data source as a DataFrame
+#'
+#' The data source is specified by the `source` and a set of options(...).
+#' If `source` is not specified, the default data source configured by
+#' "spark.sql.sources.default" will be used.
+#'
+#' @param sqlCtx SQLContext to use
+#' @param path The path of files to load
+#' @param source the name of external data source
+#' @return DataFrame
 #' @export
 #' @examples
 #'\dontrun{
 #' sc <- sparkR.init()
 #' sqlCtx <- sparkRSQL.init(sc)
-#' clearCache(sqlCtx)
+#' df <- load(sqlCtx, "path/to/file.json", source="json")
 #' }
 
-clearCache <- function(sqlCtx) {
-  callJMethod(sqlCtx, "clearCache")
+loadDF <- function(sqlCtx, path=NULL, source=NULL, ...) {
+  options <- varargsToEnv(...)
+  if (!is.null(path)) {
+    options[['path']] = path
+  }
+  sdf <- callJMethod(sqlCtx, "load", source, options)
+  dataFrame(sdf)
+}
+
+#' Create an external table
+#'
+#' Creates an external table based on the dataset in a data source,
+#' Returns the DataFrame associated with the external table.
+#'
+#' The data source is specified by the `source` and a set of options(...).
+#' If `source` is not specified, the default data source configured by
+#' "spark.sql.sources.default" will be used.
+#'
+#' @param sqlCtx SQLContext to use
+#' @param tableName A name of the table
+#' @param path The path of files to load
+#' @param source the name of external data source
+#' @return DataFrame
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' df <- sparkRSQL.createExternalTable(sqlCtx, "myjson", path="path/to/json", source="json")
+#' }
+
+createExternalTable <- function(sqlCtx, tableName, path=NULL, source=NULL, ...) {
```

Comment:
Sorry there are a few more `path=NULL` instead of `path = NULL` in this PR. Could you try to do a find and replace in the entire file ?